src/platform/notebooks/cellExecutionStateService.ts

```diff
@@ -47,18 +55,28 @@ export namespace notebookCellExecutions {
     export const onDidChangeNotebookCellExecutionState = eventEmitter.event;
 
     export function changeCellState(cell: NotebookCell, state: NotebookCellExecutionState, executionOrder?: number) {
+        const cellId = cell.metadata?.id as string | undefined;
+        const stateName = STATE_NAMES[state] || String(state);
+
+        logger.debug(
+            `[CellExecState] changeCellState called: state=${stateName}, cellId=${cellId}, executionOrder=${executionOrder}`
+        );
+
         if (state !== NotebookCellExecutionState.Idle || !executionOrder) {
+            logger.debug(`[CellExecState] Firing event immediately for state=${stateName}`);
             eventEmitter.fire({ cell, state });
             return;
         }
         // Wait for VS Code to update the cell execution state before firing the event.
+        logger.debug(`[CellExecState] Waiting for VS Code to update cell before firing Idle event`);
```

Comment:
micro nit: should we be editor agnostic? that is, avoiding using the "VS Code" name in favor of a more generic "Editor"?